packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -1113,19 +1117,25 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
   @mustCallSuper
   void handleThumbPressStart(Offset localPosition) {
     _currentController = widget.controller ?? PrimaryScrollController.of(context);
-    final Axis? direction = getScrollbarDirection();
+    final AxisDirection? direction = getScrollbarDirection();
     if (direction == null) {
       return;
     }
     _fadeoutTimer?.cancel();
     _fadeoutAnimationController.forward();
     switch (direction) {
-      case Axis.vertical:
+      case AxisDirection.down:
         _dragScrollbarAxisPosition = localPosition.dy;
         break;
-      case Axis.horizontal:
+      case AxisDirection.up:
+        _dragScrollbarAxisPosition = _flipAxis(localPosition.dy);
```

Comment:
I suggest we do all delta calculation in _updateScrollPosition so we don't need to keep track of the axis direction in handleThumbPressStart and handleThumbPressUpdate. The _updateScrollPosition can take in an Offset.

Also, do we handle corner cases like the axis direction changes during a thumb press?

Reply:
I am not sure I follow. We still need to keep track of the position where the gesture started so when we get an updated position we can make the calculation. Which Offset do you want `_updateScrollPosition` to receive?

Reply:
something like this
```dart
Offset _dragScrollbarAxisPosition;
void handleThumbPressStart(Offset localPosition) {
   _dragScrollbarAxisPosition = localPosition;
}

void handleThumbPressUpdate(Offset localPosition) {
  _updateScrollPosition(localPosition);
}

void _updateScrollPosition(Offset currentLocation) {
     double primaryDelta;
     switch (direction) {
         // calculate delta
     }
}

```

Reply:
That's what I thought. About to push an update. 😜